### DISCUSSION TITLE
Loading a trained model

### DISCUSSION THREAD

**Discussion by hananbo26** [asteroid-team/asteroid#513] [Q&A]
I have trained the Deep Clustering model but I don't know how can I load it.

I use this function to load:
model = torch_utils.load_state_dict_in(checkpoint["state_dict"], model)
and this function to get the model:
model, _ = make_model_and_optimizer(conf),
But the later requires parsing the conf.yaml file, which I fail to do.

Is there an easiest way to load a trained model? 

Thanks,

**Comment by mpariente** [comment]
For Deep Clustering, we don't have native support yet. It shouldn't be
hard, and we should have done it for a while already.

Why do you fail to load the yaml file? It's in your experiment folder.

**Reply by hananbo26** [reply]
Thanks for the answer,

I succeeded in loading the yaml file but fail to parse it. 
Attaching a screenshot of my code with the error message.

<img width="1671" alt="screenshot" src="https://user-images.githubusercontent.com/18651755/121202868-50163800-c87e-11eb-94e1-801b32ff2798.png">

**Comment by mpariente** [comment]
You should not build the parser etc.. Just load the file and instantiate
the model with it.

**Reply by hananbo26** [reply]
It works, Thanks!!!
